src/webgl/p5.RendererGL.js

```diff
@@ -1756,20 +1758,17 @@ p5.RendererGL = class RendererGL extends Renderer {
     if (this._useNormalMaterial) {
       return this._getNormalShader();
     }
-
+    // Check if the fill shader is a texture shader
     const fill = this.userFillShader;
-    if (this._enableLighting) {
-      if (!fill || !fill.isLightShader()) {
-        return this._getLightShader();
-      }
-    } else if (this._tex) {
-      if (!fill || !fill.isTextureShader()) {
-        return this._getLightShader();
-      }
-    } else if (!fill /* || !fill.isColorShader()*/) {
-      return this._getColorShader();
+    if(fill && !fill.isTextureShader()){
```

Comment:
We could change the condition to :
```suggestion
    if(fill ){
```
since, `isTextureShader()` is not checking for image. it's done in `_getImmediateFillShader()`